### PR TITLE
refactor/cut: remove unnecessary allocation and re-allocation

### PR DESCRIPTION
**Pull request by drahnr**

### CODE REVIEW

File: src/cut.rs [hunk 1]
```diff
@@ -26,10 +26,10 @@ pub fn cut_out_bytes(
     let mut out_buf = BytesMut::new();
 
     let mut remaining = size;
-    loop {
-        let mut fragment = vec![0; fragment_size-1];
```
Comment (KuabeM) [inline src/cut.rs:30]: I kept it to ensure `fragment` has the correct length when it gets filled by `to_vec()`. Without it, `to_vec()` writes the full `content` into `fragment`. Anyway, it's not a good implementation.

**Comment by KuabeM** [comment]
I added a simple solution, but I actually did not test it deeply. I'm not sure if the offsets of `split_off` are correctly set. I'll look into that either tomorrow or in the next days.

File: src/cut.rs (hunk 1 above)
Comment (drahnr) [reply src/cut.rs:30]: The thing is: the initial vector gets discarded and the vector created by `.to_vec()` is moved, so..  I doubt this ever worked as intended https://carllerche.github.io/bytes/bytes/struct.BytesMut.html#method.to_vec

Comment (drahnr) [reply src/cut.rs:30]: And the fragment size calc is off by one I think, now that I see it highlighted.

**Comment by drahnr** [comment]
~I'll have a look into it tonight too :)~

Thanks! I'll focus on getting fuzzing done for few things then.

**Comment by KuabeM** [comment]
I think we should kind of rewrite the whole `cut` module together with #7 maybe ;)

**Comment by drahnr** [comment]
:+1: Yep, I think that'd be the best approach

**Comment by KuabeM** [comment]
I went through it again and also found some bugs in the `hex_convert` module. For now, everything is fine but I think we should do the refactoring next.

**Review by KuabeM — APPROVED** [review]

**Comment by drahnr** [comment]
I think this is merge-able, afterwards I think #7 should be tackled. Any opinions?

**Pull request merged**